jsapp/js/account/security/email/emailSection.component.tsx

```diff
@@ -41,6 +46,36 @@ export default function EmailSection() {
     });
   }
 
+  function deleteNewUserEmail() {
+    deleteUnverifiedUserEmails().then(() => {
+      getUserEmails().then((data) => {
+        setEmail({
+          ...email,
+          emails: data.results,
+          newEmail: '',
+          refreshedEmail: false,
+        });
+      });
+    });
+  }
+
+  function resendNewUserEmail(unverfiedEmail: string) {
+    setEmail({
+      ...email,
+      refreshedEmail: false,
+    });
+
+    deleteUnverifiedUserEmails().then(() => {
```

Comment:
Ha I was expecting backend work here but this seems fine. The error handling is a bit trickier and we just aren't doing it right now. This is a comment and not a request for any change.